src/generateCanonicalFieldsInserts.ts

```diff
@@ -28,16 +28,27 @@ psvInput.pipe(
     delimiter: '|',
     asObject: true,
   }),
-).on('data', (row: any) => {
-  const label = row.label.replace('\'', '\'\'');
-  const shortCode = row.shortCode.replace('\'', '\'\'');
-  const dataType = row.dataType.replace('\'', '\'\'');
-  const createdAt = row.createdAt.replace('\'', '\'\'');
+).on('data', (row: Object) => {
+  const record = Object.values(row)[0];
+  //regex to remove extra quotes and correct quotes hierarchy i.e. \" inside datatype string enclosed by \' 
+  const string = record.replace(/\"/g,'\'').replace(/\'\'/g,'\"')
+  //split on the basis of comma but don't consider commas inside quotes
+  const values = string.match(/('.*?'|[^',\s]+)(?=\s*,|\s*$)/g)
+
+  const id = values[0];
```

Comment:
The team emphasized getting values by column name rather than position in the past. I think it can be done somewhat easily on the `sqlOutput.write` lines below.